worker-macros/src/durable_object.rs

```diff
@@ -109,24 +124,123 @@ pub fn expand_macro(tokens: TokenStream) -> syn::Result<TokenStream> {
 
                             #method
                         }
-                    }
+                    },
+                    "on_message" => {
+                        optional_methods.has_on_message = true;
+
+                        let mut method = impl_method.clone();
+                        method.sig.ident = Ident::new("_on_message_raw", method.sig.ident.span());
+                        method.vis = Visibility::Inherited;
+
+                        quote! {
+                            #pound[wasm_bindgen::prelude::wasm_bindgen(js_name = webSocketMessage)]
+                            pub fn _on_message(&mut self, ws: worker_sys::web_sys::WebSocket, message: String) -> js_sys::Promise {
+                                // SAFETY:
+                                // On the surface, this is unsound because the Durable Object could be dropped
+                                // while JavaScript still has possession of the future. However,
+                                // we know something that Rust doesn't: that the Durable Object will never be destroyed
+                                // while there is still a running promise inside of it, therefore we can let a reference
+                                // to the durable object escape into a static-lifetime future.
+                                let static_self: &'static mut Self = unsafe {&mut *(self as *mut _)};
+
+                                wasm_bindgen_futures::future_to_promise(async move {
+                                    static_self._on_message_raw(ws.into(), message).await.map(|_| wasm_bindgen::JsValue::NULL)
```

Comment:
According to your [docs](https://developers.cloudflare.com/durable-objects/api/websockets/#websocketmessage) this method is supposed to return `void` so upon reflection I think maybe `JsValue::UNDEFINED` is the correct type here.

However the docs also say `alert` is also `void`, but you currently have it typed as `worker::Response`, so I'm not sure what it should be.

Should it match `alert`? Or should `alert` be changed to match the documentation and return `void` / `JsValue::UNDEFINED`?